src/routes/Messages.jsx

```diff
@@ -0,0 +1,104 @@
+// React Imports
+import React, { useContext, useState, useEffect } from 'react';
+import { useLocation } from 'react-router-dom';
+// Inrupt Library Imports
+import { useSession } from '@inrupt/solid-ui-react';
+// Material UI Imports
+import Tab from '@mui/material/Tab';
+import Tabs from '@mui/material/Tabs';
+// Utility Imports
+import { getMessageTTL } from '../utils';
+// Context Imports
+import { MessageContext, SignedInUserContext } from '../contexts';
+// Component Imports
+import MessageFolder from '../components/Messages/MessageFolder';
+
+const routesArray = [{ label: 'Inbox' }, { label: 'Outbox' }];
+
+/**
+ * Messages Page - Page that generates the components for the Message system
+ * in PASS
+ *
+ * @memberof Pages
+ * @name Messages
+ * @returns {React.JSX.Element}
+ */
+
+const Messages = () => {
+  const location = useLocation();
+
+  localStorage.setItem('restorePath', location.pathname);
+
+  const { podUrl } = useContext(SignedInUserContext);
+
+  const { session } = useSession();
+  const { inboxList, setInboxList, outboxList, setOutboxList, loadMessages, setLoadMessages } =
+    useContext(MessageContext);
+
+  // Handler function for refreshing PASS messages
+  const handleMessageRefresh = async (folderType) => {
+    setLoadMessages(true);
+    const messageList = folderType === 'Inbox' ? inboxList : outboxList;
+    const messagesInSolid = await getMessageTTL(session, folderType, messageList, podUrl);
+    messagesInSolid.sort((a, b) => b.uploadDate - a.uploadDate);
+    if (folderType === 'Inbox') {
+      setInboxList(messagesInSolid);
+    } else {
+      setOutboxList(messagesInSolid);
+    }
+    setLoadMessages(false);
+  };
+
+  // Re-sorts inbox messages upon updates
```

Comment:
Can we refactor these two separate `useEffect`s into one maybe?  Not sure if this is correct, so you'd have to check, but something like this:

```js
// Re-sorts appropriate messages upon updates
useEffect(() => {
  setLoadMessages(true);
  
  // Re-sort inbox messages if inboxList changes
  if (inboxList) {
    let inboxCopy = [...inboxList];
    inboxCopy.sort((a, b) => b.uploadDate - a.uploadDate);
    setInboxList(inboxCopy);
  }

  // Re-sort outbox messages if outboxList changes
  if (outboxList) {
    let outboxCopy = [...outboxList];
    outboxCopy.sort((a, b) => b.uploadDate - a.uploadDate);
    setOutboxList(outboxCopy);
  }

  setLoadMessages(false);
}, [inboxList, outboxList]);
```

Reply:
I actually disagree. I like having them separate. It will also boost performance. Since the outbox sort will only run when the outbox updates, and the inbox sort will only run when the inbox updates. If they're combined, both sorts will happen when either are updated.

Reply:
You are correct on both parts: A) the if condition is not correct. B) probably is better separate.